### PR TITLE
575 sighting post remove unit tests

### PR DESCRIPTION
**Pull request by hwindsor**
## Pull Request Overview

- Changed existing tests that use Sighting POST to use asset_group POST
- Fixed a couplpe of bugs in the code too.

---

Application changes:
locationId is now mandatory at the Sighting level
Commit response Sighting contains fuller information, including encounters, it was just the sighting guid.

Test changes Logic:
Individuals require an Encounter, Encounters require a Sighting, Sightings require an asset group to be committed. 

Utils refactored and hopefully simplified quite a bit more. 
Asset Group create returns the response message.
Sighting create (two variants, one large, one small), extracts the guids from this (transaction, asset_group, assets, asset_group_sighting) then  commits the asset group sighting and extracts the sighting guid and the encounter guid(s) from the response. 
The return value is a dictionary of guids. 
Individual creation uses this and adds an individual on one of the encouters. It is still possible to create an individual on an existing encounter, but this is the most commonly used API. 
Again the return value is the dictionary of the guids but including the individual. 

Then each test extracts the guids it needs and as necessary, the objects. For the most part it's only the guids that are required.

### CODE REVIEW

File: tests/modules/asset_groups/resources/utils.py [hunk 1]
```diff
@@ -629,113 +628,6 @@ def commit_asset_group_sighting_sage_identification(
     return response
 
 
-def create_asset_group_and_sighting(
```
Comment (naknomum) [inline tests/modules/asset_groups/resources/utils.py:632]: there it goes!  :wave:

File: app/modules/asset_groups/metadata.py [hunk 2]
```diff
@@ -204,7 +204,7 @@ def validate_encounters(cls, encounters, debug):
     def _validate_sighting(self, sighting, file_dir, sighting_debug, encounter_debug):
 
         sighting_fields = [
-            ('locationId', str, False),
+            ('locationId', str, True),
```
Comment (hwindsor) [inline app/modules/asset_groups/metadata.py:207]: It was mandatory in the Sighting create so I made this mandatory here too.

File: app/modules/asset_groups/resources.py [hunk 3]
```diff
@@ -480,7 +480,7 @@ class AssetGroupSightingCommit(Resource):
     )
     # NOTE this returns a sighting schema not an AssetGroup one as the output of this is that
     # a sighting is created
-    @api.response(BaseSightingSchema())
+    @api.response(DetailedSightingSchema())
```
Comment (hwindsor) [inline app/modules/asset_groups/resources.py:483]: Previouusly the response just contained the sighting guid. This seemed to be a far more sensible option (and made the tests easier too)

File: tests/modules/social_groups/resources/test_social_group.py [hunk 4]
```diff
@@ -8,35 +8,68 @@
...
 import tests.modules.individuals.resources.utils as individual_utils
+import tests.modules.sightings.resources.utils as sighting_utils
 import tests.modules.audit_logs.resources.utils as audit_utils
 from tests import utils as test_utils
 
 log = logging.getLogger(__name__)
 
 
-def create_individuals(db, flask_app_client, user, request, num_individuals=3):
+# Needs to be a separate method as if you call the addfinalizer in a loop, it only deleted the last one
```
Comment (colinwkingen) [inline tests/modules/social_groups/resources/test_social_group.py:18]: Good to know

**Review by colinwkingen — APPROVED** [review]
LGTM, will wait for parallel review from Jon before merge

**Review by naknomum — APPROVED** [review]
looks good.  lotta work, thanks howard.

**Pull request merged**